contracts/SBCDepositContract.sol

```diff
@@ -224,4 +242,183 @@ contract SBCDepositContract is IDepositContract, IERC165, IERC677Receiver, Pausa
+        require(success, "Withdrawal processing failed");
+        if (amountToProceed == failedWithdrawalRecord.amount) {
+            failedWithdrawalRecord.processed = true;
+        } else {
+            failedWithdrawalRecord.amount -= amountToProceed;
+        }
+        emit FailedWithdrawalProcessed(_failedWithdrawalId, amountToProceed, failedWithdrawalRecord.receiver);
+    }
+
+    uint256 public failedWithdrawalsPointer;
+
+    /**
+     * @dev Function to be used to process failed withdrawals.
+     * Call to this function will revert only if it ran out of gas.
+     * Call to this function doesn't transmit flow control to any untrusted contract,
+     * so using constant gas limit and constant max number of withdrawals for calls of this function is ok.
+     * @param _maxNumberOfFailedWithdrawalsToProcess Maximum number of failed withdrawals to be processed.
+     */
+    function processFailedWithdrawalsFromPointer(uint256 _maxNumberOfFailedWithdrawalsToProcess) external {
+        for (uint256 i = 0; i < _maxNumberOfFailedWithdrawalsToProcess; ++i) {
+            if (failedWithdrawalsPointer == numberOfFailedWithdrawals) {
+                break;
+            }
+
+            FailedWithdrawalRecord storage failedWithdrawalRecord = failedWithdrawals[failedWithdrawalsPointer];
+            if (!failedWithdrawalRecord.processed) {
+                bool success = _processWithdrawal(
+                    failedWithdrawalRecord.amount,
+                    failedWithdrawalRecord.receiver,
+                    onWithdrawalsUnwrapToGNOByDefault
+                );
+                if (!success) {
+                    break;
+                }
+                failedWithdrawalRecord.processed = true;
+                emit FailedWithdrawalProcessed(
+                    failedWithdrawalsPointer,
+                    failedWithdrawalRecord.amount,
+                    failedWithdrawalRecord.receiver
+                );
+            }
+
+            ++failedWithdrawalsPointer;
+        }
+    }
+
+    /**
+     * @dev Function to be used only in the system transaction.
+     * Call to this function will revert only in three cases:
+     *     - the caller is not `SYSTEM_WITHDRAWAL_EXECUTOR` or `_admin()`;
+     *     - the length of `_amounts` array is not equal to the length of `_addresses` array;
+     *     - the call ran out of gas.
+     * Call to this function doesn't transmit flow control to any untrusted contract,
+     * so using constant gas limit and constant number of withdrawals for calls of this function is ok.
+     * @param _amounts Array of amounts to be withdrawn.
+     * @param _addresses Array of addresses that should receive the corresponding amount of tokens.
+     */
+    function executeSystemWithdrawals(uint64[] calldata _amounts, address[] calldata _addresses) external {
+        require(
+            _msgSender() == SYSTEM_WITHDRAWAL_EXECUTOR || _msgSender() == _admin(),
```

Comment:
Why also allow for _admin address?

Reply:
That is more for test purposes. I mean anyway the admin have rights to do everything with this contract, but this way we have the opportunity to conveniently test this contract.

Reply:
Has the function name defined in the [spec](https://github.com/gnosischain/specs/blob/master/execution/withdrawals.md) been changed permanently?

Reply:
No. I just used more declarative one name. For sure we can change it if it is required.

Reply:
@Barichek I like this better as I prefer the function name to be a verb. So, either this one or just `withdraw`. And since this function is called from the execution client, I need to know what name to use. If we stick with this one, the spec needs to be updated.

Reply:
I can update the spec with `executeSystemWithdrawals`. While admin can do anything, having it allowed here explicitly is a bit unclean. Can you find another approach to become the system executor on testing?

Reply:
Please let me help